### PR TITLE
Fix: use VARCHAR instead of TEXT for MySQL pks, indexed fields

### LINKED ISSUE #1204: MySQL cannot index text fields
MySQL cannot create an index or primary key on columns that are of type text, they need to be VARCHAR with a fixed length.

Currently the migrations don't work on MySQL because we create indices on name which is text. Perhaps we need to do some kind of if statement inside migrations to make limit the name to varchar(...).

### PR DESCRIPTION
**Pull request by georgesittas**
Fixes #1204

### CODE REVIEW

File: sqlmesh/utils/migrations.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from sqlglot.dialects.dialect import DialectType
```
Comment (izeigerman) [inline sqlmesh/utils/migrations.py:1]: [Nit] Can we not call the module in plural form?

File: sqlmesh/utils/migrations.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+# https://dev.mysql.com/doc/refman/8.0/en/storage-requirements.html#data-types-storage-reqs-strings
+MAX_TEXT_LENGTH = 65535
```
Comment (izeigerman) [inline sqlmesh/utils/migrations.py:4]: This seems a little extreme and will impact the page layout in mysql. Can we put something more reasonable in here? I'm sure something like 1024 or even 512 will be more than enough.

Comment (GeorgeSittas) [reply sqlmesh/utils/migrations.py:4]: Fair enough!

Comment (GeorgeSittas) [reply sqlmesh/utils/migrations.py]: Changed to 1024, left a comment to make it clear that it might not match MySQL's limit.

**Review by tobymao — APPROVED** [review]

File: sqlmesh/migrations/v0008_create_intervals_table.py [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
         intervals_table,
         {
             "id": exp.DataType.build("text"),
             "created_ts": exp.DataType.build("bigint"),
-            "name": exp.DataType.build("text"),
-            "identifier": exp.DataType.build("text"),
-            "version": exp.DataType.build("text"),
+            "name": exp.DataType.build(pk_text_type),
+            "identifier": exp.DataType.build(pk_text_type),
+            "version": exp.DataType.build(pk_text_type),
```
Comment (vchan) [inline sqlmesh/migrations/v0008_create_intervals_table.py:22]: Please ignore this if I'm mistaken, but isn't the primary key for this table `id` and so `id`'s type needs to be updated but not these columns?

Comment (tobymao) [reply sqlmesh/migrations/v0008_create_intervals_table.py:22]: everything that has an index needs to be varchar fixed

File: sqlmesh/utils/migration.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+from sqlglot.dialects.dialect import DialectType
+
+# This is an ad-hoc upper bound which doesn't necessarily match MySQL's
+# TEXT storage specification, but it should be good for most use cases.
+MAX_TEXT_LENGTH = 1024
+
+
+def primary_key_text_type(dialect: DialectType) -> str:
```
Comment (tobymao) [inline sqlmesh/utils/migration.py:8]: change the name of this, it's not primary key, it's index

File: sqlmesh/migrations/v0008_create_intervals_table.py (hunk 3 above)
Comment (vchan) [reply sqlmesh/migrations/v0008_create_intervals_table.py:22]: Thank you for clarifying. So does `id` also need to have its type updated to varchar?

Comment (tobymao) [reply sqlmesh/migrations/v0008_create_intervals_table.py:22]: the problem is that mysql doesn't allow unbounded chars in indices

Comment (vchan) [reply sqlmesh/migrations/v0008_create_intervals_table.py:22]: Sorry, I didn't mean just varchar, but the varchar(1024) in the pk_text_type variable. This create table will fail in MySQL if the `id` column is type text and it's used in the primary key index, right?

Comment (tobymao) [reply sqlmesh/migrations/v0008_create_intervals_table.py:22]: yes @GeorgeSittas you need to update "id"

Comment (GeorgeSittas) [reply sqlmesh/migrations/v0008_create_intervals_table.py:22]: You're right @vchan, thanks for catching this!

File: sqlmesh/utils/migration.py (hunk 4 above)
Comment (GeorgeSittas) [reply sqlmesh/utils/migration.py]: Done

**Pull request merged**